packages/mermaid/src/diagrams/git/parser/gitGraph.jison

```diff
@@ -109,10 +110,17 @@ branchStatement
 
 cherryPickStatement
     : CHERRY_PICK COMMIT_ID STR {yy.cherryPick($3, '', undefined)}
+    | CHERRY_PICK COMMIT_ID STR PARENT_COMMIT STR {yy.cherryPick($3, '', undefined,$5)}
     | CHERRY_PICK COMMIT_ID STR COMMIT_TAG STR {yy.cherryPick($3, '', $5)}
-    | CHERRY_PICK COMMIT_ID STR COMMIT_TAG EMPTYSTR {yy.cherryPick($3, '', '')}
+    | CHERRY_PICK COMMIT_ID STR PARENT_COMMIT STR COMMIT_TAG STR {yy.cherryPick($3, '', $7,$5)}
+    | CHERRY_PICK COMMIT_ID STR COMMIT_TAG STR PARENT_COMMIT STR {yy.cherryPick($3, '', $5,$7)}
```

Comment:
I know this is working and we are heading towards another parser, but if there is one more option to this then we probably have to create grammar rules for all the permutations.

If arguments order is the only difference between those repetitive lines I would split them into several grammar rules like this (just an example):

```
cherryPickStatement: CHERRY_PICK commit (cherry_pick_argument)
commit
: COMMIT_TAG [STR | EMPTY_STR]
| COMMIT_ID STR

cherry_pick_argument
: PARENT_COMMIT STR { $$={parentCommit: $STR} }
| COMMIT_TAG [STR | EMPTY_STR]
| COMMIT_ID STR
|
```
etc. Assigning result to $$ will form ascending tree. This way we could avoid bothering about arguments order. But since the whole previous grammar is excessive and redundant I think this is OK.

Reply:
@RounakJoshi09 let the question be resolved, but if you would like to improve it, feel free to open new issue and PR